### DISCUSSION TITLE
Invisible signature field and PDF locking

### DISCUSSION THREAD

**Discussion by joowdx** [lsnepomuceno/laravel-a1-pdf-sign#160] [Q&A]
Hi, is there a way to make the `setImage()` appear as not invisible when opening the pdf with Acrobat?

Also can we lock the file after it has been signed?

**Comment by lsnepomuceno** [comment]
Hello @joowdx,

Two questions, and they have different answers — one is solved, the other is honestly not implemented.

## The visible appearance: yes

In 2.0 the seal is a first-class part of the builder, and it renders a visible appearance in Acrobat:

```php
use LSNepomuceno\LaravelA1PdfSign\Data\SealPlacement;
use LSNepomuceno\LaravelA1PdfSign\Enums\FontSize;
use LSNepomuceno\LaravelA1PdfSign\Facades\A1PdfSign;

$signed = A1PdfSign::newSignature()
    ->certificate($pfxPath, $password)
    ->pdf($pdfPath)
    ->seal(
        placement: new SealPlacement(x: 155, y: 250, width: 50, page: SealPlacement::LAST_PAGE),
        fontSize: FontSize::Large,
        showExpiry: true,
    )
    ->sign();

// or stamp an image you already have, skipping the renderer
->sealFrom('path/to/seal.png')
```

The seal is drawn onto the signature field's widget annotation, so a reader shows it as the signature's appearance rather than as a loose image. Omitting `seal()` gives an invisible signature, which is still cryptographically valid — the appearance is not part of the cryptography.

Position, size, page, driver, font and colour all come from configuration now, instead of being hard-coded as they were in 1.x.

## Locking the document: no, and I would rather say so plainly

**The package does not implement certification signatures.** Every signature it writes is an *approval* signature: there is no `/DocMDP` transform and no `/Perms` entry anywhere in the codebase, so nothing instructs a reader to forbid changes after signing.

What you get instead is **detection, not prevention**:

```php
$report = A1PdfSign::validate($path);
$report->isValid();   // false if the covered bytes changed
```

Each signature covers the document as it stood at its own revision, so any later modification is detectable — a reader will show the signature as invalid, or as valid-but-with-subsequent-changes. That is a different guarantee from locking, and for many workflows it is the one that actually matters, but it is not what you asked for and I do not want to imply otherwise.

Worth noting that the reason multiple signatures work at all in 2.0 is that signing appends a revision rather than rebuilding the document — and document locking is in tension with that, since a certification signature with `/DocMDP 1` forbids the later revisions that additional signatures need.

If certification/DocMDP is what you need, please open an issue saying so. It is a plausible addition — the writer already produces the revision structure it would attach to — and knowing there is a real use case is what would move it up.
